distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Helpers.enso

```diff
@@ -0,0 +1,17 @@
+from Standard.Base import all
+
+import project.Any.Any
+import project.Data.Locale.Locale
+import project.Data.Text.Case_Sensitivity.Case_Sensitivity
+import project.Errors.Illegal_Argument.Illegal_Argument
+import project.Meta
+
+## PRIVATE
+
+   Assert that `text_maybe` is a Text, then call the action.
+expect_text : Any -> Any -> Any ! Illegal_Argument
+expect_text text_maybe ~action = case text_maybe of
+    _ : Text -> action
+    _ ->
+        msg = "Expected Text, got " + (Meta.type_of text_maybe . to_display_text)
+        Error.throw (Illegal_Argument.Error msg)
```

Comment:
Interesting. I think it's indeed worth making this check although throughout the library we are usually not checking the types as thoroughly. The idea is that a typechecker at some point should do the job for us, but that will probably not happen anytime soon.
I think we need to have a good balance. Often we do not check. I think this check is OK to keep.

I'm just wondering if it's worth changing `Illegal_Argument` to `Type_Error` here.

Reply:
Changed to Type_Error.